File: VonexDealer.Vue/ClientApp/models/ISDN.js
export class ISDN {
    constructor({
        isdnid = 0,
        landlineID = 0,
        contractLengthMonths = 0,
        ratePlanID = 0,
        hasRange = false,
        rangeSize = null,
        addressID = 0,
        numbers = [],
        isdnType = null
    } = {}) {
        this.isdnid = isdnid;
        this.landlineID = landlineID;
        this.contractLengthMonths = contractLengthMonths;
        this.ratePlanID = ratePlanID;
        this.hasRange = hasRange;
        this.rangeSize = rangeSize;
        this.addressID = addressID;
        this.numbers = numbers;
        this.isdnType = isdnType;
            
    }

}
export function createISDN(data) {
    return Object.freeze(new ISDN(data));
}

File: VonexDealer.API/Models/Order/ISDN.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace VonexDealer.API.Models.Order
{
    public class ISDN
    {
        public Int64 ISDNID { get; set; }
        public Int64 LandlineID { get; set; }

        public int ContractLengthMonths { get; set; }
        public Int64 RatePlanID { get; set; }
        public bool HasRange { get; set; }
        public int RangeSize { get; set; }
        public Int64 AddressID { get; set; }
        public string NumbersAsString { get; set; }
        [NotMapped]
        public List<String> Numbers
        {
            get
            {
                if (NumbersAsString != null)
                    return NumbersAsString.Split(',').ToList();

                return null;

            }
            set
            {
                NumbersAsString = String.Join(",", value);
            }
        }

    }
}